driver:audio: sf32lb52x:Add onchip codec driver for sf32lb52x platform

ljq-ctrl opened the pull request.

erwango committed:
boards: *: Fix wrong feature identifier for usb_device

`usb device` isn't a valid feature name, fix it, using the new "usbd" tag.
Remove it when not enabled on the board.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

maje-emb committed:
modules: nordic: select correct timer for NFCT on nRF54L

Set the appropriate default NFC timer for the nRF54L series SoCs.

Ref: NCSDK-31244

Signed-off-by: Marcin Jelinski <marcin.jelinski@nordicsemi.no>

mausys committed:
soc: xlnx: zynq7000: move SoC code to common folder

remove duplicate xc7zxxx and xc7zxxxs soc code

Signed-off-by: Simon Maurer <mail@maurer.systems>

ljq-ctrl committed:
sifli sf32lb audio codec driver

ttwards committed:
drivers: dma: stm32: add null callback checking

Fix the STM32 DMA driver did not check if the optional dma_callback was
set in dma_stm32_configure(). This could lead to a hard fault when an
interrupt occurs and the callback is NULL, even though the API allows
the callback to be omitted.

Fixes #97454
Signed-off-by: Wenxi Xu <xuwenxi0517@gmail.com>

ljq-ctrl committed:
Merge branch 'zephyrproject-rtos:main' into main

ZhaoxiangJin commented on drivers/audio/Kconfig:
```diff
@@ -12,6 +12,9 @@ if AUDIO
 # Audio Codec options
 #
+
+source "drivers/audio/Kconfig.sf32lb"
```
This is not the correct position, it should be placed [here](https://github.com/zephyrproject-rtos/zephyr/pull/97906/commits/2eb4f502bc924f7b7f14152889da2715fc631af3#diff-fb6f7b4daa158ebeb4aacd8b870963ca91608feb2f63712b3e0c863402022416R49).

ZhaoxiangJin commented on drivers/audio/sf32lb_codec.c:
```diff
@@ -0,0 +1,820 @@
+	.start = sf32lb_codec_start,
+	.stop = sf32lb_codec_stop,
+	.set_dac_volume = sf32lb_codec_set_dac_volume,
+	.set_dac_mute = sf32lb_codec_set_dac_mute,
```
Is there a driver API named `set_dac_volume`?

ljq-ctrl replied:
self defined API in sf32lb_codec.h

nordic-bami committed:
tests: drivers: flash: Add test with buffer size over the bus packet limit

Applicable to flash devices with MSPI controller

Signed-off-by: Bartosz Miller <bartosz.miller@nordicsemi.no>

ZhaoxiangJin commented:
Please take a look at https://docs.zephyrproject.org/latest/contribute/guidelines.html, your commit message does not comply with Zephyr conventions.

ZhaoxiangJin commented on drivers/audio/Kconfig.sf32lb:
```diff
@@ -0,0 +1,11 @@
+
+config AUDIO_CODEC_SF32LB
+	bool "sf32lb codec driver support"
+	select USE_SIFLI_HAL
```
what is ‘sco’ ?

gmarull commented on drivers/audio/sf32lb_codec.c:
```diff
@@ -0,0 +1,820 @@
+
+extern void set_pll_freq_type(uint8_t type);
+extern void HAL_TURN_ON_PLL();
+extern void HAL_TURN_OFF_PLL();
```
definitely -1, we need clock control driver to be extended.

ljq-ctrl replied:
fixed

gmarull commented on drivers/audio/sf32lb_codec.c:
```diff
@@ -0,0 +1,820 @@
+#define AUDCODEC_DAC0_DMA_IRQ               DMAC1_CH1_IRQn
+#define AUDCODEC_DAC0_DMA_IRQ_PRIO          0
+#define AUDCODEC_DAC0_DMA_INSTANCE          DMA1_Channel1
+#define AUDCODEC_DAC0_DMA_REQUEST           41
```
-1, this needs to be properly described in devicetree, use DMA driver

gmarull reviewed (changes requested):
This PR has some fundamental issues, see my comments. Also please, do not introduce platform specific samples for drivers that are supposed to offer a vendor-agnostic interface. Only use these samples if you have to extend standard interface with vendor-specific extra APIs.

ljq-ctrl committed:
Merge remote-tracking branch 'upstream/main'

mathieuchopstm committed:
drivers: usb: udc: stm32: use addr_before_status capability

Instead of snooping SETUP packets for SET_ADDRESS request to call the HAL
SetAddress() function manually, mark device with the addr_before_status
capability such that the UDC stack will call udc_stm32_set_address()
before submitting status by itself.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

HoZHel committed:
drivers: entropy: Fix non-stop RNG ISR firing for STM32WB09

Fix the TRNG driver issue regarding non-stop ISR firing for STM32WB09 by
clearing RNG_IRQ_SR_ERROR_IRQ flag.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

HoZHel committed:
drivers: bluetooth: hci: Fix the issue about the TRNG peripheral instance

Fix the issue regarding passing the TRNG peripheral instance to the driver.

Increase the SYSTEM_WORKQUEUE_STACK_SIZE when CONFIG_BT is set.
According to the log of thread analyzer for beacon sample, 1048 bytes
are needed. So, it's been increased to a safer value.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

pdgendt committed:
scripts: Bump ruff to 0.14.2

Update the ruff version to the latest 0.14.2 and update the excluded files
to match with the last checks for linter and formatting issues.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

JiafeiPan committed:
cmake: toolchain: cross_compile: check newlib and picolibc

Add support to check whether Cross Compile toolchain support newlib and
picolibc or not.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

khoatranyj committed:
tests: subsys: fs: Update test configuration for Renesas RA

Remove redundant "partition" and "slot1_partition" nodes that are not
needed and conflict with the MRAM device node labeled "mram" on
Renesas RA boards. These conflicts caused build errors.

For the affected board overlays, the SDHC and SDMMC nodes are now
enabled at the board or shield layer, so they no longer need to be
defined in the test configuration.

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

khoatranyj committed:
tests: drivers: i2c: Fix incorrect file paths in SCI I2C test support

The current paths to the .overlay and .conf files used for SCI I2C tests
are incorrect, as they include {NORMALIZED_BOARD_QUALIFIERS}, which does
not exist in the actual file names.

This commit corrects the file paths in the test support configuration.

Signed-off-by: Khoa Tran <khoa.tran.yj@bp.renesas.com>

nordicjm committed:
doc: release: migration_guide: 4.3: Add notes on TF-M and MCUboot

Adds notes of changes made to these

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: releases: release_notes: 4.3: Add new MCUmgr Kconfigs

Lists new Kconfigs for MCUmgr

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

modersohn committed:
boards: stm32n6570_dk: fix overlapping partitions

* reducing mcuboot partition size so slot_0 and slot_1 can stay where
  they are
* fixes #97850

Signed-off-by: Sebastian Modersohn <im24@msohn.de>

Ayush1325 committed:
dts: vendor: ti: k3-am62-main: Change license to Apache-2

This probably should have been caught much earlier, but well, here we
are. The original license came from the following linux dt that
BeagleBoard.org uses [0]. However, due to differences between linux and
zephyr dt and the fact that zephyr does not yet support most of the
devices on this soc, the only lines same between the two files are the
`status = "disabled";`, and some of the address (not all since things
like GPIO are handled differently). Keeping that in mind, this file
should never have used that license or have TI Copyright.

If it's too late to change the license now, we can probably add this
file to exceptions, since the current license of the file, i.e. GPL-2.0
and MIT should cover all cases where Apache-2 can be used. Again,
apologies since there was no reason for this file to use the dual
license.

[0]:https://github.com/beagleboard/BeagleBoard-DeviceTrees/blob/v6.12.x-Beagle/src/arm64/ti/k3-am62-main.dtsi

Signed-off-by: Ayush Singh <ayush@beagleboard.org>

AllenZhang-NXP committed:
drivers: flash: Added clear_flash_caches function for MCXW2XX

MCXW2XX ROMAPI provide the function FLASH_CacheClear to clear the cache.
The SMSCM is not supported on MCXW2XX platform.

Signed-off-by: Allen Zhang <chunfeng.zhang@nxp.com>

AllenZhang-NXP committed:
soc: mcxw2xx: Add clock enablement for flash controller

kCLOCK_Sysctl must be enabled for FLASH_CacheClear,
FLASH_CacheSpeculationControl and FLASH_CheckECC to have an effect.

Signed-off-by: Allen Zhang <chunfeng.zhang@nxp.com>

rugeGerritsen committed:
soc: nordic: nrf54h: s2ram: Fix compiling with FPU

This commit fixes a change introduced in #97025
where too many definitions where removed.

Fixes issue #98382

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

natto1784 committed:
samples: adc_stream: fix sample filter

io-channels in zephyr,user is required for this sample to compile, so
add that to the filter and remove the redundant ifdef in the source code.

Signed-off-by: Amneesh Singh <amneesh@ti.com>

JordanYates committed:
wifi: nrf_wifi: named MAC address choice

Add a name to the "Wi-Fi MAC address type" choice symbol so that the
default can be updated on a per-board basis.

Signed-off-by: Jordan Yates <jordan@embeint.com>

HoZHel committed:
drivers: entropy: Modify the recover_seed_error function for WB0x

Exclude the WB0x series from discarding data 12 times from the RNG data
register.

Add a comment to the recover_seed_error function explaining the reason
for reading and discarding data from the RNG data register 12 times.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

nngt88 committed:
drivers: bluetooth: h4: Fix check for sufficient buffer size

When alloc the evt buffer,such as the adv report, only compare the
remaining data len, should aslo consider the hdr_len, because the
hdr also copy to alloced buffer.if not consider the hdr, then
hdr + remaining data may larger than alloced buffer, because the
alloced buffer is not enough,then will assert when receive the
remaining data.

Signed-off-by: Guotao Zhang <guotao.zhang@nxp.com>

carlescufi committed:
scripts: compliance: Harden the execution of npx

Instead of relying on various exceptions that can happen when one tries
to execute a program that is not installed in the system (typically
FileNotFoundError but sometimes PermissionError), use shutil.which() to
ensure the npx executable is available before running it.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

sjanc committed:
Bluetooth: Host: Fix crash on bt_disable() with limited advertising

When limited advertising is enabled there is pending deleyable work
for timing it out. If in such case struct bt_le_ext_adv is cleared by
memset system will crash on next tick.

Fix this by ensuring work is cancelled before clearing struct
bt_le_ext_adv. Using bt_le_ext_adv_foreach() ensures that this will
be correctly handled with and without extended advertising enabled.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

williamtcdns committed:
xtensa: suppress warning on variable used uninitialized

This change suppress the warning:
variable 'ie' is used uninitialized whenever switch default is
taken [-Wsometimes-uninitialized]

Signed-off-by: William Tambe <williamt@cadence.com>

williamtcdns committed:
xtensa: properly compute irq number

This is a miss from:
abeccfec289 xtensa: support for more than 32 interrupts

Signed-off-by: William Tambe <williamt@cadence.com>

carlescufi committed:
kernel: events: Depend on multithreading

Kernel events depend on multithreading being enabled, and mixing them
with a non-multithreaded build gives linker failures internal to
events.c. To avoid this, make events depend on multithreading.

```
libkernel.a(events.c.obj): in function `k_event_post_internal':
175: undefined reference to `z_sched_waitq_walk'
events.c:183: undefined reference to `z_sched_wake_thread'
events.c:191: undefined reference to `z_reschedule'
libkernel.a(events.c.obj): in function `k_sched_current_thread_query':
kernel.h:216: undefined reference to `z_impl_k_sched_current_thread_query'
libkernel.a(events.c.obj): in function `k_event_wait_internal':
events.c:312: undefined reference to `z_pend_curr'
```

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

kartben committed:
samples: instrumentation: remove unnecessary include of instrumentation.h

The instrumentation.h header is not needed to compile the
instrumentation sample and it can be misleading for folks to think they
have to include it in their code to use the subsystem, so remove it.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: releases: introduce release notes and migration guide docs for 4.4

This introduces the release notes and migration guide for Zephyr 4.4.0
earlier so that people have a placeholder to start adding content as
they line up pull requests for the 4.4 release.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

ljq-ctrl committed:
Merge remote-tracking branch 'upstream/main'

ljq-ctrl committed:
driver:audio: sf32lb52x:Add onchip codec driver for sf32lb52x platform #97905

Signed-off-by: Jiqun Li <jiqunli@sifli.com>

ck-telecom committed:
drivers: dma: sf32lb: fix dma compile error

Fix dma clock compile error

Signed-off-by: Qingsong Gou <gouqs@hotmail.com>

ljq-ctrl committed:
driver:audio: sf32lb52x:Add onchip codec driver for sf32lb52x platform #97905

Signed-off-by: Jiqun Li <jiqunli@sifli.com>

JordanYates committed:
modules: nrf_wifi: add missing Kconfig dependency

`ZVFS_OPEN_ADD_SIZE_NRF70_ENABLE_DUAL_VIF` should only exist in the
build if `NRF70_ENABLE_DUAL_VIF`, not for every application in tree.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
tests: lib: fdtable: declare the fd used by test

Ensure that the test has a free file descriptor to use.

Signed-off-by: Jordan Yates <jordan@embeint.com>

AllenZhang-NXP committed:
drivers: flash: Fixed the typo in clear_flash_caches

elif should be else

Signed-off-by: Allen Zhang <chunfeng.zhang@nxp.com>

JarmouniA commented on drivers/audio/Kconfig.sf32lb:
```diff
@@ -0,0 +1,10 @@
+	select USE_SIFLI_HAL
+	select DMA
+	default y
+	depends on SOC_SERIES_SF32LB52X
```
```suggestion
	depends on DT_HAS_SIFLI_SF32LB_ENABLED
```

JarmouniA reviewed (changes requested):
merge commits are not allowed, see https://docs.zephyrproject.org/latest/contribute/guidelines.html#pull-request-guidelines

jukkar committed:
tests: posix: xsi_realtime: Set enough file descriptors

Earlier fix in commit bfd033df81a82960cc5a99f731acc931c504339a
caused the Posix xsi_realtime test to fail. Set minimum amount
of file descriptors in the test to pass the build assert.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: dns_resolver: The test needs more fds

Set minimum amount of file descriptors in the test to pass
the tests.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: lwm2m_engine: The test needs more fds

Set minimum amount of file descriptors in the test to pass
the tests.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: posix: portability: fs: Increase number of fds

The test had too few file descriptors.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

gmarull reviewed (changes requested):
Please make sure to adhere to commit/contributing guidelines.

ljq-ctrl committed:
driver:audio: sf32lb52x:Add onchip codec driver for sf32lb52x platform #97906

support audio playback and capure for sf32lb52x platform, extend codec.h API

Signed-off-by: Jiqun Li <jiqunli@sifli.com>

JarmouniA commented:
Please take a look at https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow and fix your git branch.

JarmouniA commented:
@ljq-ctrl Please leave comments resolving to reviewers.

ljq-ctrl commented:
Encountered an unsolvable problem using GitHub for the first time,  close this PR
recommit with this new PR

driver:audio: sf32lb52x:Add onchip codec driver for sf32lb52x platfor… #98701
